databricks/sdk/mixins/workspace.py

```diff
@@ -8,6 +10,54 @@
 def _fqcn(x: any) -> str:
     return f'{x.__module__}.{x.__name__}'
 
+_LOG = logging.getLogger('databricks.sdk')
+
+class _ParallelRecursiveListing:
+    def __init__(self, path, listing, threads, notebooks_modified_after):
+        self.path = path
+        self.listing = listing
+        self.threads = threads
+        self.notebooks_modified_after = notebooks_modified_after
+        self.directories = Queue()
+        self.results = Queue()
+        self.directories.put_nowait(path)
+        self._start()
```

Comment:
Recommend against starting from the constructor, can start from `__iter__`.

Reply:
yep, it's a PoC still